site/app/models/gradeable/LateDays.php

```diff
@@ -43,24 +43,10 @@ public function __construct(Core $core, User $user, array $graded_gradeables, $l
         $graded_gradeables = array_filter($graded_gradeables, function (GradedGradeable $gg) {
             return $gg->getGradeable()->getType() === GradeableType::ELECTRONIC_FILE;
         });
-        $ggs = [];
-        foreach ($graded_gradeables as $gg) {
-            $ggs[$gg->getGradeableId()] = $gg;
-        }
+
         // Get the late day updates that the instructor will enter
         $this->late_days_updates = $late_day_updates ?? $this->core->getQueries()->getLateDayUpdates($user->getId());
-        $this->core->getQueries()->generateLateDayCacheForUser($user->getId());
         $late_day_cache = $this->core->getQueries()->getLateDayCacheForUser($user->getId());
-        // Construct late days info for each gradeable
```

Comment:
Don't we need to use getLateDayCacheForUserGradeable function instead to replace line 55-63, which calls generateLateDayCacheForUser function inside.